components/brave_new_tab_ui/api/initialData.ts

```diff
@@ -43,14 +43,16 @@ export async function getInitialData (): Promise<InitialData> {
       topSites,
       defaultSuperReferralTopSites,
       brandedWallpaperData
-    ] = await Promise.all([
-      preferencesAPI.getPreferences(),
-      statsAPI.getStats(),
-      privateTabDataAPI.getPrivateTabData(),
-      topSitesAPI.getTopSites(),
-      !isIncognito ? brandedWallpaper.getDefaultSuperReferralTopSites() : Promise.resolve(undefined),
-      !isIncognito ? brandedWallpaper.getBrandedWallpaper() : Promise.resolve(undefined)
-    ])
+    ] = await preferencesAPI.getPreferences().then((preferences: preferencesAPI.Preferences) => {
```

Comment:
Please can we not mix `await` and `.then`?  Since this is "sync" you can just await the preference retrieval first.

Reply:
why use `Promise.all` then?

Reply:
Because `Promise.all` executes all the asynchronous operations in parallel, which we can do with non inter-dependent operations. Am I understanding the question right @AndriusA?